core/src/main/java/org/kiwix/kiwixmobile/core/search/SearchActivity.kt

```diff
@@ -155,6 +155,10 @@ class SearchActivity : BaseActivity() {
     searchViewModel.actions.offer(OnItemClick(it))
   }
 
+  private fun onItemClickNewTab(it: SearchListItem) {
+    searchViewModel.actions.offer(OnItemClick(it, true))
```

Comment:
We shouldn't be reusing this action, create a new action to represent that the user has clicked this new icon

Reply:
@macgills Is the code duplication caused by making a new version of the saveSearchAndOpenItem() function (in SearchViewModel.kt) okay? I would need to make a new function because the parameter type OnItemClick would need to be changed to OnItemClickNewTab.

Reply:
it is fine, create a new action, handle it and push the parameter into the function to reuse it
```
  private fun saveSearchAndOpenItem(it: OnItemClick, openInNewTab:Boolean) {
    effects.offer(
      SaveSearchToRecents(recentSearchDao, it.searchListItem, zimReaderContainer.id)
    )
    effects.offer(
      OpenSearchItem(it.searchListItem, openInNewTab)
    )
  }
```

Reply:
Yeah I was thinking of doing it like that, but the saveSearchAndOpenItem function expects the "it" parameter to be of type "OnItemClick", and for the new tab version, "it" would actually be of type "OnOpenInNewTabClick". So it would give an error, unless I casted it, which I'm not sure is valid.

Reply:
Instead of  OnItemClick pass a searchListItem, both OnItemClick & OnOpenInNewTabClick  will have 1

Reply:
Also for code in our ViewModels/SideEffects it would be great if you could add unit tests

Reply:
Thanks, I'll do that.

Reply:
Ah, I didn't pay enough attention to your layout file, my apologies, I only have so much time a day on this project.

Instead of using an include we can just use a textview here